src/AdoNetCore.AseClient/AseDataReader.cs

```diff
@@ -10,24 +10,45 @@ namespace AdoNetCore.AseClient
 {
     public sealed class AseDataReader : DbDataReader
     {
-        private readonly IEnumerator<TableResult> _results;
+        private TableResult _currentTable;
+        private readonly BlockingCollection<TableResult> _results;
         private int _currentResult = -1;
         private int _currentRow = -1;
         private readonly CommandBehavior _behavior;
+        private bool _hasFirst;
 
 #if ENABLE_SYSTEM_DATA_COMMON_EXTENSIONS
         private readonly AseCommand _command;
         private DataTable _currentSchemaTable;
 #endif
 
-        internal AseDataReader(IEnumerable<TableResult> results, AseCommand command, CommandBehavior behavior)
+        internal AseDataReader(AseCommand command, CommandBehavior behavior)
         {
-            _results = results.GetEnumerator();
+            _results = new BlockingCollection<TableResult>();
```

Comment:
I'd say put a comment in saying that the default underlying collection is FIFO (`ConcurrentQueue`)

Reply:
Have backed this out.